streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/NamedSuppressed.java

```diff
@@ -0,0 +1,10 @@
+package org.apache.kafka.streams.kstream.internals.suppress;
+
+/**
+ * Internally-facing interface to work around the fact that all Suppressed config objects
+ * are name-able, but do not present a getter (for consistency with other config objects).
+ * If we allow getters on config objects in the future, we can delete this interface.
+ */
+public interface NamedSuppressed {
```

Comment:
Just a side remark: I guess with KIP-307 this can be removed, too?

Reply:
If it's an internal interface, might it make sense to `NamedSuppressed extends Suppressed` such that both internal sub-classed inherit only NamedSuppressed (ie, this insures that we don't forget to inherit both interfaces?) -- just a thought -- it's of course less compossible than.

Reply:
Yep, KIP-307 would make this redundant as well.

You have a good point about making it extend suppressed; I didn't put too much thought into it, as it's just a temporary band-aid with a really limited surface area.

I'll make this change.